docs/content/basic/authenticating-oauth.md

```diff
@@ -4,256 +4,703 @@ lang: en
+[generating a unique authorization URL](#extra-authorizations).
 
 :::
 
----
-## Redirect URI
-Bolt for JavaScript provides a **Redirect URI Path** `/slack/oauth_redirect`. Slack uses the Redirect URI to redirect users after they complete an app's installation flow. 
+### Redirect URL
 
-You will need to add the full **Redirect URI** including your app domain in your Slack app configuration settings under **OAuth and Permissions**, e.g. `https://example.com/slack/oauth_redirect`. 
+Bolt for JavaScript provides the **Redirect URL** path `/slack/oauth_redirect`
+out-of-the-box for Slack to use when redirecting users that complete the OAuth
+installation flow.
 
-To supply your own custom **Redirect URI**, you can set `redirectUri` in the App options and `installerOptions.redirectUriPath`. You must supply both, and the path must be consistent with the full URI.
+You will need to add the full **Redirect URL** including your app domain in
+[app settings][settings] under **OAuth and Permissions**, e.g.
+`https://example.com/slack/oauth_redirect`.
+
+To supply a custom Redirect URL, you can set `redirectUri` in the App options
+and `installerOptions.redirectUriPath`. Both must be supplied and be consistent
+with the full URL if a custom Redirect URL is provided:
 
 ```javascript
 const app = new App({
   signingSecret: process.env.SLACK_SIGNING_SECRET,
   clientId: process.env.SLACK_CLIENT_ID,
   clientSecret: process.env.SLACK_CLIENT_SECRET,
-  stateSecret: 'my-state-secret',
-  scopes: ['chat:write'],
-  redirectUri: 'https://example.com/slack/redirect', // here
+  stateSecret: "my-state-secret",
+  scopes: ["chat:write"],
+  // highlight-next-line
+  redirectUri: "https://example.com/slack/redirect",
   installerOptions: {
-    redirectUriPath: '/slack/redirect', // and here!
+    // highlight-next-line
+    redirectUriPath: "/slack/redirect",
   },
 });
 ```
 
----
+#### Custom callbacks
 
-## Installation object
-Bolt will pass your `installationStore`'s `storeInstallation` handler an `installation`. This can be a source of confusion for developers who aren't sure what shape of object to expect. The `installation` object should resemble:
+The page shown after OAuth is complete can be changed with
+`installerOptions.callbackOptions` to display different details:
+
+```javascript
+const app = new App({
+  signingSecret: process.env.SLACK_SIGNING_SECRET,
+  // ...
+  installerOptions: {
+    // highlight-start
+    callbackOptions: {
+      success: (installation, installOptions, req, res) => {
+        res.send("The installation succeeded!");
+      },
+      failure: (error, installOptions, req, res) => {
+        res.send("Something strange happened...");
+      },
+    },
+    // highlight-end
+  },
+});
+```
+
+Full reference reveals [these additional options][callback-options] but if no
+options are provided, the [`defaultCallbackSuccess`][callback-default-success]
+and [`defaultCallbackFailure`][callback-default-failure] are used.
```

Comment:
hmmm i want to make these have nouns attached to them but not sure what is right. "callback"?